tests/integration/pilot/tunneling/forward-proxy/configmap.tmpl.yaml

```diff
@@ -0,0 +1,105 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: external-forward-proxy-config
```

Comment:
I don't love deploying a whole new envoy with custom code for this... seems like it may be problematic to maintain. Can we use the existing gateway and an EnvoyFilter?

Reply:
The test checks if connection came from forward proxy IP to make sure, that tunneling was properly performed, so if egress gateway would act as forward proxy, then it couldn't be checked whether connection came from egress gateway or whether from forward proxy.